File: language-support/js/daml-ledger/src/validation/get_transactions_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { object, Validation, native } from "./base";
import { OptionalValidation, RequiredValidation } from "./base/typelevel";
import { LedgerOffset } from "./ledger_offset";
import { TransactionFilter } from "./transaction_filter";

function required(): RequiredValidation<ledger.GetTransactionsRequest> {
    return {
        begin: LedgerOffset,
        filter: TransactionFilter
    };
}

function optional(): OptionalValidation<ledger.GetTransactionsRequest> {
    return {
        end: LedgerOffset,
        verbose: native('boolean')
    };
}

export const GetTransactionsRequest: Validation = object<ledger.GetTransactionsRequest>('GetTransactionsRequest', required, optional);

File: sdk/canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/transaction/GetTransactionsRequest.scala
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.transaction

import com.digitalasset.canton.data.Offset
import com.digitalasset.canton.ledger.api.TransactionFilter

final case class GetTransactionsRequest(
    startExclusive: Option[Offset],
    endInclusive: Option[Offset],
    filter: TransactionFilter,
    verbose: Boolean,
)
